src/types.ts

```diff
@@ -52,28 +59,44 @@ export interface ISession {
   state: string;
 }
 
-export type Emailer = string;
-
 export type EmailHeader = string;
 
 export type Attachment = File;
 
-export interface IEmail {
+export interface IEmailProperties {
   id: string;
   blobId: string;
   threadId: string;
   mailboxIds: { [key: string]: boolean };
-  keywords: { [key: string]: boolean };
-  from: Emailer[] | null;
-  to: Emailer[] | null;
+  keywords: IEmailKeywords;
+  from: IEmailAddress[] | null;
+  to: IEmailAddress[] | null;
   subject: string;
-  date: Date;
   size: number;
   preview: string;
   attachments: Attachment[] | null;
   createdModSeq: number;
   updatedModSeq: number;
-  deleted: Date | null;
+  receivedAt: IUtcDate;
+}
+
+export type IUtcDate = string;
+export type ITrue = true;
+
+export interface IEmailKeywords {
+  $draft?: ITrue;
+  $seen?: ITrue;
+  $flagged?: ITrue;
+  $answered?: ITrue;
+  $forwarded?: ITrue;
+  $phishing?: ITrue;
+  $junk?: ITrue;
+  $notjunk?: ITrue;
```

Comment:
I'm not fluent in JavaScript but I got the feeling we only offer a close set of Keyworks.

JMAP allows passing arbitrary keywords. `Set<String>` is likely a better data structure to represent it (convenience methods can be exposed)

At least it can come in follow up refactorings.

Reply:
Those keywords are related only to Email entity https://jmap.io/spec-mail.html#metadata as far as I can tell.

However:
> Users may add arbitrary keywords to an Email. For compatibility with IMAP, a keyword is a case-insensitive string of 1–255 characters in the ASCII subset %x21–%x7e (excludes control chars and space), and it MUST NOT include any of these characters

I would assume, that it could be added in future changes. Why I am offering it at the moment? Due to it still being in early stages it is easier to limit our logic to use of `interfaces`, `enums` and `types`. If we start to introduce classes it might become a bit more complex.

Due to JSON structure being returned the way it is (regarding keywords), we can't set it as `Set` without some conversion logic.